### PR TITLE
Update GitHub release job to include plugin artifacts

### PR DESCRIPTION
**Pull request by buildbreaker**
Resolves https://github.com/bufbuild/connect-kotlin/issues/4

Adding the protoc-gen-connect-kotlin jar to the releases.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -41,3 +43,13 @@ jobs:
           export ORG_GRADLE_PROJECT_signingInMemoryKey=$(gpg --armor --passphrase $GPG_PASSPHRASE --pinentry-mode=loopback --export-secret-keys $GPG_KEY_NAME signing-key | grep -v '\-\-' | grep -v '^=.' | tr -d '\n')
           shred signing-key
           make release
+      - name: sha256
+        run: sha256sum ./protoc-gen-connect-kotlin/build/libs/protoc-gen-connect-kotlin.jar >> sha256.txt
+      - name: Upload release artifacts in GitHub
```
Comment (rebello95) [inline .github/workflows/release.yml:48]: Might want to change this to something like `Publish release` since this will actually create a release

**Review by rebello95 — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (buildbreaker) [reply .github/workflows/release.yml:48]: Yeah there's two publishing going on here though: the sonatype/maven publish and this github one.

Comment (buildbreaker) [reply .github/workflows/release.yml:48]: Publish GitHub artifacts?

Comment (rebello95) [reply .github/workflows/release.yml:48]: Sure, the main thing I'm saying is that it creates a GitHub release too, and then pushes the artifacts with the release

**Pull request merged**